File: Frontend/controller/orderController.js
window.orderController = function ($scope, $http, $location, $routeParams) {
    // $scope.title = "Product detail"
   
    $http.get("http://localhost:8080/order/get-pdList")
    $http.get("http://localhost:8080/order/show").then(function (response) {
        if (response.status == 200) {
            $scope.cartList = response.data.cartList;
            $scope.billList = response.data.billList;
            $scope.customerList = response.data.customerList;
            $scope.productDetailsList = response.data.productDetailsList;
            $scope.total = response.data.total;
            
        } else {
            console("Ket noi that bai")
        }
    })



    $scope.getQuantity = function (id, quantity) {
        $http.get("http://localhost:8080/order/get-quantity/" + id + "/" + quantity)
        $http.get("http://localhost:8080/order/show").then(function (response) {
            if (response.status == 200) {
                $scope.cartList = response.data.cartList;
                $scope.productDetailsList = response.data.productDetailsList;
                $scope.total = response.data.total;
            } else {
                console("Ket noi that bai")
            }
        })
        
        $('#exampleModal' + id).modal('hide');
    }

    $scope.editQuantity = function (id, quantity) {
        $http.get("http://localhost:8080/order/edit-quantity/" + id + "/" + quantity)
        $http.get("http://localhost:8080/order/show").then(function (response) {
            if (response.status == 200) {
                $scope.cartList = response.data.cartList;
                $scope.productDetailsList = response.data.productDetailsList;
                $scope.total = response.data.total;
            } else {
                console("Ket noi that bai")
            }
        })
        $('#suaSoLuong' + id).modal('hide');
    }

    $scope.removeProduct = function (id) {
        $http.get("http://localhost:8080/order/remove-product/" + id)
        $http.get("http://localhost:8080/order/show").then(function (response) {
            if (response.status == 200) {
                $scope.cartList = response.data.cartList;
                $scope.productDetailsList = response.data.productDetailsList;
                $scope.total = response.data.total;

            } else {
                console("Ket noi that bai")
            }
        })
    }

    $scope.createBill = function () {
        $http.get("http://localhost:8080/order/add").then(function (response) {
            if (response.status == 200) {
                $scope.billList = response.data;
            } else {
                console("Ket noi that bai")
            }
        })
    }

    $scope.selectBill = function (id) {
        $http.get("http://localhost:8080/order/select-bill/" + id).then(function(response){
            if (response.status == 200) {
                $scope.detailBill = response.data;
            } else {
                console("Ket noi that bai")
            }
        })
    }

    $scope.validateForm = function () {
        let check = true
        $scope.checkBillSelect = {
            id: false,
            total:false
        }

        if (!$scope.detailBill || !$scope.detailBill.id) {
            check = false;
            $scope.checkBillSelect.id = true
        }

        if ($scope.total == 0.00 || !$scope.total) {
            check = false;
            $scope.checkBillSelect.total = true
        }
        return check;
    }

    $scope.pay = function () {
        if ($scope.validateForm()) {
            let bill = {
                id:$scope.detailBill.id,
                diaChi: $scope.detailBill.diaChi == "" ? null:$scope.detailBill.diaChi,
                soDienThoai:$scope.detailBill.soDienThoai == "" ? null:$scope.detailBill.soDienThoai,
            }

            $http.post("http://localhost:8080/order/pay/"+$scope.detailBill.id, bill).then(function (response) {
                if (response.status == 200) {
                    alert("Add success!")
                    $http.get("http://localhost:8080/order/show").then(function (response) {
                        if (response.status == 200) {
                            $scope.cartList = response.data.cartList;
                            $scope.billList = response.data.billList;
                            $scope.customerList = response.data.customerList;
                            $scope.productDetailsList = response.data.productDetailsList;
                            $scope.total = null;
                            $scope.detailBill = null;
                        } else {
                            console("Ket noi that bai")
                        }
                    })
                } else {
                    alert("Add failed!")
                }
            })
        }
    }
}

File: Backend/src/main/java/com/example/demo/controller/OrderController.java
package com.example.demo.controller;

import com.example.demo.entity.Bill;
import com.example.demo.service.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/order")
public class OrderController {
    @Autowired
    OrderService orderService;

    Boolean isReset = false;
    @GetMapping("/show")
    public Map<String, Object> show() {
        orderService.getData();
        if (isReset){
            orderService.getDataProductDetails();
            isReset = false;
        }
        Map<String, Object> data = new HashMap<>();
        data.put("cartList",orderService.cartList);
        data.put("billList",orderService.billList);
        data.put("customerList",orderService.customerList);
        data.put("total",orderService.total);
        data.put("productDetailsList",orderService.productDetailsList);
        return data;
    }


    @GetMapping("/get-pdList")
    public void getPDList(){
        orderService.clearCart();
        isReset = true;
    }
    @GetMapping("/get-quantity/{id}/{quantity}")
    public void getQuantity(@PathVariable Long id,@PathVariable int quantity){
        orderService.getQuantity(quantity,id);
    }

    @GetMapping("/remove-product/{id}")
    public void removeProduct(@PathVariable Long id){
        orderService.removeProductInCart(id);
    }

    @GetMapping("/edit-quantity/{id}/{quantity}")
    public void editQuantity(@PathVariable Long id,@PathVariable int quantity){
        orderService.editQuantity(quantity,id);
    }

    @GetMapping("/select-bill/{id}")
    public Bill selectBill(@PathVariable Long id){
        return orderService.selectBill(id);
    }

    @GetMapping("/add")
    public List<Bill> addBill(){
        return orderService.add();
    }

    @PostMapping("/pay/{id}")
    public void pay(@PathVariable Long id, @RequestBody Bill bill){
        orderService.pay(id,bill);
    }


}
